py4j-java/src/py4j/commands/Command.java

```diff
@@ -88,9 +88,10 @@ public void execute(String commandName, BufferedReader reader,
 	 * <p>
 	 * Called when a command instance is created and assigned to a connection.
 	 * </p>
-	 * 
+	 *
 	 * @param gateway
+	 * @param connection the {@link GatewayConnection} this socket is assigned to
 	 */
-	public void init(Gateway gateway);
+	public void init(Gateway gateway, GatewayConnection connection);
```

Comment:
can we add an init method instead of changing the signature and breaking backward compatibility?